src/ray/util/event.cc

```diff
@@ -148,26 +148,51 @@ void EventManager::ClearReporters() { reporter_map_.clear(); }
 ///
 thread_local std::unique_ptr<RayEventContext> RayEventContext::context_ = nullptr;
 
+std::unique_ptr<RayEventContext> RayEventContext::first_instance_ = nullptr;
+
+std::atomic<int> RayEventContext::first_instance_started_setting_(0);
+
+std::atomic<bool> RayEventContext::first_instance_finished_setting_(false);
+
 RayEventContext &RayEventContext::Instance() {
   if (context_ == nullptr) {
     context_ = std::unique_ptr<RayEventContext>(new RayEventContext());
   }
   return *context_;
 }
 
+RayEventContext &RayEventContext::OriginContext() {
+  if (first_instance_finished_setting_ == false) {
+    static RayEventContext tmp_instance_;
+    return tmp_instance_;
+  }
+  return *first_instance_;
+}
+
 void RayEventContext::SetEventContext(
     rpc::Event_SourceType source_type,
     const std::unordered_map<std::string, std::string> &custom_fields) {
   source_type_ = source_type;
   custom_fields_ = custom_fields;
+
+  initialized_ = true;
+  if (!first_instance_started_setting_.fetch_or(1)) {
+    first_instance_ = std::unique_ptr<RayEventContext>(new RayEventContext());
```

Comment:
prefer make_unique instead of new. https://stackoverflow.com/questions/37514509/advantages-of-using-stdmake-unique-over-new-operator